src/main/java/com/jcabi/ssh/SSH.java

```diff
@@ -215,7 +237,15 @@ protected Session session() throws IOException {
                 CharEncoding.UTF_8
             );
             jsch.setHostKeyRepository(new EasyRepo());
-            jsch.addIdentity(file.getAbsolutePath());
+            if (this.passphrase == null) {
```

Comment:
@LinkedList let's add a puzzle to change the design later to get rid of nullability.
Actually, the whole design with abstract class is not the best. It's better to decorate and compose